pkg/container-utils/docker/docker.go

```diff
@@ -46,10 +47,16 @@ type DockerClient struct {
 	socketPath string
 }
 
-func NewDockerClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
+func NewDockerClient(socketPath string, useCri bool) (runtimeclient.ContainerRuntimeClient, error) {
 	if socketPath == "" {
 		socketPath = runtimeclient.DockerDefaultSocketPath
 	}
+	if useCri {
+		// TODO: Configurable
+		socketPath = runtimeclient.CriDockerDefaultSocketPath
+		criClient, err := cri.NewCRIClient(types.RuntimeNameDocker, socketPath, DefaultTimeout)
+		return &criClient, err
+	}
```

Comment:
I'm wondering if this should be managed outside of the docker and containerd files. IMO, it's strange to have the `New` function here, but the rest of the implementation in the CRI file. It might be confusing for the developers in the future. Maybe it should be managed in `NewContainerRuntimeClient`. Something like this:

```go
	case types.RuntimeNameDocker:
		socketPath := runtime.SocketPath

		if runtime.Extra.RuntimeProtocol == types.RuntimeProtocolCRI {
			if envsp := os.Getenv("INSPEKTOR_GADGET_DOCKER_CRI_SOCKETPATH"); envsp != "" && socketPath == "" {
				socketPath = filepath.Join(host.HostRoot, envsp)
			}
			return runtimeclient.NewDockerCRIClient(socketPath)
		}

		if envsp := os.Getenv("INSPEKTOR_GADGET_DOCKER_INTERNAL_SOCKETPATH"); envsp != "" && socketPath == "" {
			socketPath = filepath.Join(host.HostRoot, envsp)
		}
		return docker.NewDockerClient(socketPath, runtime.Extra.UseCri)
	case types.RuntimeNameContainerd:
		socketPath := runtime.SocketPath
		if envsp := os.Getenv("INSPEKTOR_GADGET_CONTAINERD_SOCKETPATH"); envsp != "" && socketPath == "" {
			socketPath = filepath.Join(host.HostRoot, envsp)
		}

		if runtime.Extra.RuntimeProtocol == types.RuntimeProtocolCRI {
			return runtimeclient.NewContainerdCRIClient(socketPath)
		}

		return containerd.NewContainerdClient(socketPath, runtime.Extra)
```

Then, in `cri,go`:

```go
func NewCRIDockerClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
	if socketPath == "" {
		// Docker uses different socket paths for CRI and internal Docker protocol.
		socketPath = runtimeclient.CriDockerDefaultSocketPath
	}
	return NewCRIClient(types.RuntimeNameDocker, socketPath, DefaultTimeout)
}

func NewCRIContainerdClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
	if socketPath == "" {
		// Containerd uses the same socket for both CRI and internal containerd protocol.
		socketPath = runtimeclient.ContainerdDefaultSocketPath
	}
	return NewCRIClient(types.RuntimeNameContainerd, socketPath, DefaultTimeout)
}
```

BTW, probably all the `os.Getenv(...)` call should be done inside the `New` functions.